File: app/src/main/java/com/zj/test/baseloadingview/MainActivity.java
package com.zj.test.baseloadingview;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.zj.loading.ZLoadingView;
import com.zj.loading.DisplayMode;
import com.zj.loading.OnTapListener;
import com.zj.loading.OverLapMode;
import com.zj.test.R;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private ZLoadingView<?, ?, ?> blvView;
    private int lapModeId = 0;
    private OverLapMode curOverlapMod = OverLapMode.FO;
    private Toast toast = null;
    private int index;
    private final List<DisplayMode> mode = new ArrayList<>();
    private final OverLapMode[] modes = new OverLapMode[]{
            OverLapMode.OVERLAP, OverLapMode.FLOATING, OverLapMode.FO
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        blvView = findViewById(R.id.bld_view);
        Button btView = findViewById(R.id.bt_view);
        ImageView ivBg = findViewById(R.id.iv_bg);
        TextView tvOverride = findViewById(R.id.tv_override);

        blvView.setMode(DisplayMode.LOADING.delay(1000));

        mode.add(0, DisplayMode.LOADING);
        mode.add(1, DisplayMode.NO_DATA);
        mode.add(2, DisplayMode.NO_NETWORK);
        mode.add(3, DisplayMode.NORMAL);

        ivBg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.e("zj --- ", "bg.onclick");
            }
        });
        blvView.setOnTapListener(new OnTapListener() {
            @Override
            public void onTap() {
                if (toast == null)
                    toast = Toast.makeText(MainActivity.this, "", Toast.LENGTH_SHORT);
                toast.show();
                toast.setText("call refresh with error");
            }
        });
        btView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                DisplayMode mod = mode.get(index);
                blvView.setMode(mod, curOverlapMod);
                index++;
                if (index == mode.size()) index = 0;
            }
        });
        tvOverride.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                curOverlapMod = modes[lapModeId % 3];
                ++lapModeId;
            }
        });
    }
}

File: app/src/main/java/com/zj/test/baseloadingview/MainActivity.kt
package com.zj.test.baseloadingview

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.Toast
import com.zj.loading.BaseLoadingView
import com.zj.test.R

class MainActivity : AppCompatActivity() {

    private var btView: Button? = null
    private var ivBg: ImageView? = null
    private var blvView: BaseLoadingView? = null
    private var cbOverride: CheckBox? = null
    private var isOverrideBg = true

    private var toast: Toast? = null
    private var index: Int = 0

    private var hints = HashMap<BaseLoadingView.DisplayMode, String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test)
        blvView = findViewById(R.id.bld_view)
        btView = findViewById(R.id.bt_view)
        ivBg = findViewById(R.id.iv_bg)
        cbOverride = findViewById<CheckBox>(R.id.cb_override)

        hints[BaseLoadingView.DisplayMode.loading] = getString(R.string.loading)
        hints[BaseLoadingView.DisplayMode.noData] = getString(R.string.noData)
        hints[BaseLoadingView.DisplayMode.noNetwork] = getString(R.string.noNetwork)
        hints[BaseLoadingView.DisplayMode.normal] = ""

        ivBg?.setOnClickListener {
            Log.e("zj --- ", "bg.onclick")
        }

        blvView?.setRefreshListener {
            if (toast == null) toast = Toast.makeText(this@MainActivity, "", Toast.LENGTH_SHORT)
            toast?.show()
            toast?.setText("call refresh with error")
        }
        btView?.setOnClickListener {
            val mode = BaseLoadingView.DisplayMode.values()[index]
            blvView!!.setMode(mode, hints[mode], isOverrideBg)
            index++
            if (index == BaseLoadingView.DisplayMode.values().size) index = 0
        }
        cbOverride?.setOnCheckedChangeListener { _, isCheck ->
            isOverrideBg = !isCheck
        }
    }
}
